src/Manager/ClientFilter.php

```diff
@@ -11,17 +11,17 @@
 final class ClientFilter
 {
     /**
-     * @var Grant[]
+     * @var list<Grant>
      */
     private $grants = [];
 
     /**
-     * @var RedirectUri[]
+     * @var list<RedirectUri>
      */
     private $redirectUris = [];
 
     /**
-     * @var Scope[]
+     * @var list<Scope>
```

Comment:
`list` type is supported by psalm, phpstan, phan, and phpstorm, should we use it? or should we rollback to phpstorm generic array?

Reply:
I'm learning here :) It feels more self-explanatory/safe and I guess it is, since you added it. 
Let's keep it as-is and experiment (for me at least ^^)

Reply:
the difference:

- foo[] is array<array-key, foo> ( array<string|int, foo> )
- list<foo> is array<int, foo> and keys are always orders ( basically a vector )

Reply:
thanks for confirming!